### PR TITLE
Rate limit endpoint enum

### PR DESCRIPTION
**Pull request by marcogruhl**
All Endpoints decorated with
- Description
- Rate Limits (per App/User, reset Time)
- Counts against the Tweet Cap
- max operations limits
And also Endpoint relevant information
- Endpoint Type
- Endpoint Url
- Access Level

This information will help to get relations between these properties. 

All information are carried together from:
[Twitter API v2](https://developer.twitter.com/en/docs/twitter-api)
[Developer Portal - Products](https://developer.twitter.com/en/portal/products/)
[Rate limits](https://developer.twitter.com/en/docs/twitter-api/rate-limits)

### CODE REVIEW

File: src/Response/RRateLimit/AccessLevel.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace TwitterSharp.Response.RRateLimit
+{
+    public enum AccessLevel
```
Comment (Xwilarg) [inline src/Response/RRateLimit/AccessLevel.cs:3]: You may want to use the [Flag](https://docs.microsoft.com/en-us/dotnet/api/system.flagsattribute?view=net-6.0) attribute on that

**Review by Xwilarg — CHANGES_REQUESTED** [review]

File: src/Response/RRateLimit/AccessLevel.cs (hunk 1 above)
Comment (marcogruhl) [reply src/Response/RRateLimit/AccessLevel.cs:3]: Don't know about that - changed it

**Comment by marcogruhl** [comment]
I also optimized the Namespaces. Now i it suits better. I cant name the Endpoint-"Folder" "Endpoint" because it collides with the enum Name. So ApiEndpoint was my solution to that.

**Review by Xwilarg — CHANGES_REQUESTED** [review]

File: src/Response/RateLimit.cs [hunk 2]
```diff
@@ -1,13 +1,15 @@
-﻿namespace TwitterSharp.Response
+﻿using TwitterSharp.ApiEndpoint;
+
+namespace TwitterSharp.Response.RRateLimit
```
[the remark this answers is not in the document]
Comment (marcogruhl) [reply src/Response/RateLimit.cs]: sorry, i missed that - fixed
the refactoring also found a "wrong" namespace in answer.cs ... refactored that too ... please verify

**Review by Xwilarg — APPROVED** [review]

**Pull request merged**